x/ibc/20-transfer/keeper/keeper_test.go

```diff
@@ -1,24 +1,53 @@
 package keeper_test
 
 import (
+	"fmt"
 	"testing"
 
-	"github.com/stretchr/testify/suite"
-
-	abci "github.com/tendermint/tendermint/abci/types"
-
 	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/simapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/x/ibc/20-transfer/keeper"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	connection "github.com/cosmos/cosmos-sdk/x/ibc/03-connection"
+	channel "github.com/cosmos/cosmos-sdk/x/ibc/04-channel"
+	"github.com/cosmos/cosmos-sdk/x/ibc/20-transfer/types"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// define constants used for testing
+const (
+	testChainID    = "test-chain-id"
+	testClient     = "test-client"
+	testClientType = clientexported.Tendermint
+
+	testConnection = "testconnection"
+	testPort1      = "firstport"
+	testPort2      = "secondport"
+	testChannel1   = "firstchannel"
+	testChannel2   = "secondchannel"
+
+	testChannelOrder   = channel.ORDERED
+	testChannelVersion = "1.0"
+)
+
+// define variables used for testing
+var (
+	testAddr1 = sdk.AccAddress([]byte("testaddr1"))
+	testAddr2 = sdk.AccAddress([]byte("testaddr2"))
+
+	testCoins, _          = sdk.ParseCoins("100atom")
+	testPrefixedCoins1, _ = sdk.ParseCoins(fmt.Sprintf("100%satom", types.GetDenomPrefix(testPort1, testChannel1)))
+	testPrefixedCoins2, _ = sdk.ParseCoins(fmt.Sprintf("100%satom", types.GetDenomPrefix(testPort2, testChannel2)))
 )
 
 type KeeperTestSuite struct {
 	suite.Suite
 
-	cdc    *codec.Codec
-	ctx    sdk.Context
-	keeper *keeper.Keeper
+	cdc *codec.Codec
+	ctx sdk.Context
+	app *simapp.SimApp
```

Comment:
Why do you need the app instead of just the keeper?

Reply:
> Why do you need the app instead of just the keeper?

It's convenient to use the app to verify the packet proof with the next block header.